### PR TITLE
Populate Test Suite Attributes

### PR DESCRIPTION
**Pull request by FuadBalashov**
The specification for the testsuites tag in Junit includes multiple attributes, as per: http://llg.cubic.org/docs/junit/
This commit populates disabled, errors, failures, tests and time.
It does so by recursively collecting those metrics from nested testsuite tags in `jasmineDone` and then passing them to `wrapOutputAndWriteFile`.

### CODE REVIEW

**Comment by FuadBalashov** [comment]
Hello! As described in the commit description I have added logic to include test failures, tests, errors and time in the testsuites tag. This more closely matches the structure of the linked junit schema and makes it easier to have other infrastructure read the generated xml files for the number of failures and disabled tests.

@bloveridge @larrymyers I am not sure who maintains this repo so I tagged both of you. Please let me know how to proceed!

**Comment by FuadBalashov** [comment]
Oh, also it looks like some tests are failing with the newest version of node (8) which came out 2 days ago. I think this is an issue in master and I don't think I have touched any code related to them.

File: src/junit_reporter.js [hunk 1]
```diff
@@ -306,6 +312,27 @@
             }
         };
 
+        self.formatSuiteData = function(suite) {
+            return {
+                disabled: suite._disabled || 0,
+                errors: 0,
```
Comment (FuadBalashov) [inline src/junit_reporter.js]: on second thought it probably makes sense to remove errors to reflect the structure of the inner testsuite tags.

**Comment by FuadBalashov** [comment]
ping: @bloveridge @larrymyers

**Comment by FuadBalashov** [comment]
ping: @bloveridge @larrymyers

**Comment by FuadBalashov** [comment]
ping: @bloveridge @larrymyers

**Comment by kraenhansen** [comment]
@FuadBalashov I can verify the same tests failing on https://github.com/larrymyers/jasmine-reporters/pull/170 unrelated to the changes that I've made.

**Comment by FuadBalashov** [comment]
ping: @bloveridge @larrymyers

**Comment by Brian-Kaplan** [comment]
Any movement on this?

**Comment by bloveridge** [comment]
Thanks for the patch, and sorry for the abysmally slow response.

**Comment by FuadBalashov** [comment]
Thank you @bloveridge! Any prediction on when the next release will be?

**Comment by bloveridge** [comment]
It will be within the next week. There is still one more pull request that might get updated and merged in that time. Look for the new reporters version by Dec 1.

**Comment by FuadBalashov** [comment]
Hey @bloveridge it looks like there hasn't been a release yet. Do you have an update?

**Comment by bloveridge** [comment]
Just published `2.3.0`. Sorry for the delay.

**Comment by FuadBalashov** [comment]
Thank you, looking forward to using the new release

**Pull request merged**